src/Model/Item.php

```diff
@@ -60,4 +62,143 @@ public static function update(array $fields, array $condition)
 
 		return $rows;
 	}
+
+	/**
+	 * @brief Add a shadow entry for a given item id that is a thread starter
+	 *
+	 * We store every public item entry additionally with the user id "0".
+	 * This is used for the community page and for the search.
+	 * It is planned that in the future we will store public item entries only once.
+	 *
+	 * @param integer $itemid Item ID that should be added
+	 */
+	public static function addShadow($itemid)
+	{
+		$fields = ['uid', 'wall', 'private', 'moderated', 'visible', 'contact-id', 'deleted', 'network', 'author-id', 'owner-id'];
+		$condition = ["`id` = ? AND (`parent` = ? OR `parent` = 0)", $itemid, $itemid];
+		$item = dba::selectFirst('item', $fields, $condition);
+
+		if (!DBM::is_result($item)) {
+			return;
+		}
+
+		// is it already a copy?
+		if (($itemid == 0) || ($item['uid'] == 0)) {
+			return;
+		}
+
+		// Is it a visible public post?
+		if (!$item["visible"] || $item["deleted"] || $item["moderated"] || $item["private"]) {
+			return;
+		}
+
+		// is it an entry from a connector? Only add an entry for natively connected networks
+		if (!in_array($item["network"], [NETWORK_DFRN, NETWORK_DIASPORA, NETWORK_OSTATUS, ""])) {
+			return;
+		}
+
+		// Is the public contact configured as hidden?
+		if (Contact::isHidden($item["owner-id"]) || Contact::isHidden($item["author-id"])) {
+			return;
+		}
+
+		// Only do these checks if the post isn't a wall post
+		if (!$item["wall"]) {
+			// Check, if hide-friends is activated - then don't do a shadow entry
+			if (dba::exists('profile', ['is-default' => true, 'uid' => $item['uid'], 'hide-friends' => true])) {
+				return;
+			}
+
+			// Check if the contact is hidden or blocked
+			if (!dba::exists('contact', ['hidden' => false, 'blocked' => false, 'id' => $item['contact-id']])) {
+				return;
+			}
+		}
+
+		// Only add a shadow, if the profile isn't hidden
+		if (dba::exists('user', ['uid' => $item['uid'], 'hidewall' => true])) {
+			return;
+		}
+
+		$item = dba::selectFirst('item', [], ['id' => $itemid]);
+
+		if (count($item) && ($item["allow_cid"] == '')  && ($item["allow_gid"] == '') &&
```

Comment:
Please use `DBM::is_result` instead of `count()`

Reply:
Sorry, I forgot this.